src/storage/ResourceStore.ts

```diff
@@ -4,6 +4,55 @@ import type { RepresentationPreferences } from '../http/representation/Represent
 import type { ResourceIdentifier } from '../http/representation/ResourceIdentifier';
 import type { Conditions } from './Conditions';
 
+/**
+ * The types of modification being tracked
+ */
+export type ModificationType = 'CREATED' | 'CHANGED' | 'DELETED';
+
+/**
+ * A modified resource with a specific modification type
+ */
+export interface ModifiedResource {
+  resource: ResourceIdentifier;
+  modificationType: ModificationType;
+}
```

Comment:
I only quickly skimmed so this is not a full review. But I noticed this.

When the `ResourceStore` functions where changed to return changed identifiers it was already considered to return metadata instead of arrays: #632 . This might be a good time to do this instead of inventing a completely new type just for this.

Reply:
The first. `ResourceIdentifier[]` was a temporary solution until something more extensive was needed.

Reply:
Alright then I understand, I admit I would need to look into how the `RepresentationMetadata` can be used for this. Let's talk about this next Thursday on the architecture meeting. Thank you.

Reply:
If we do end up using metadata I would no longer use an enum for the modification type but define the possible values as named nodes in a vocabulary (see [Vocabularies.ts](https://github.com/solid/community-server/blob/main/src/util/Vocabularies.ts)).

To store it in the metadata I would use an rdf format like
```turtle
<resource> solid-as:activity as:Created, as:Changed.
```
with `as` being the the activity streams vocabulary and `solid-as` being an internal custom vocabulary. This is not exactly how that vocabulary is supposed to be used but it does make our life easier and is fine (in my opinion) as long as this is used internally. (I randomly picked the name `solid-as:activity`, can be improved).

Assuming we do that, these could then easily be added to a metadata object using `metadata.add(SOLID_AS.activity, AS.Created);` with SOLID_AS and AS being imports from `Vocabularies.ts`.

Reply:
When multiple resources have changed in one operation(e.g. the parent container was recursively created during creation of a specific resource), shall all these changes be represented as a single `RepresentationMetadata` object, or would you rather prefer an array(e.g. `RepresentationMetadata[]`)?

Reply:
> When multiple resources have changed in one operation

That's a good point, I forgot this was a possibility. Thinking out loud here, both have some advantages and disadvantages. Putting everything in 1 metadata makes it so there is always 1 response object which is nice. The disadvantage is that this way there would be metadata about different resources in a single object. With an array the issue might be that if we return extra metadata besides this we might not know which object we have to look into. 

My first thought is that the array solution will probably make it clearer what is going on.

Reply:
I've begun refactoring this PR to eliminate the `ModifiedResource` class and `ModificationType` enum.
I'll update this PR once the unit tests are satisfied. 👍

Reply:
Array or Map of resource and metadata?

Reply:
True, map would also be an option, with the key being the resource identifier. I don't really have a strong opinion here either way since currently you would have to iterate through all the metadata objects being returned.

Reply:
A `Map` of representation metadatas sounds quite appealing to me since as far as I know the order of entries doesn't matter, and the Map provides a nice and quick lookup.

Reply:
I would use a simple object and not an actual `Map` though. So a `Record<string, RepresentationMetadata>`.